UserExperience_LazyLoad_GoogleMaps_GoogleMapsEasy.php

```diff
@@ -1,68 +1,126 @@
 <?php
+/**
+ * File: UserExperience_LazyLoad_GoogleMaps_GoogleMapsEasy.php
+ *
+ * @package W3TC
+ */
+
 namespace W3TC;
 
+/**
+ * Class UserExperience_LazyLoad_GoogleMaps_GoogleMapsEasy
+ */
 class UserExperience_LazyLoad_GoogleMaps_GoogleMapsEasy {
+	/**
+	 * Stores the URL of the script to be preloaded.
+	 *
+	 * @var string
+	 */
 	private $preload_url = '';
 
-
-
+	/**
+	 * Modifies the buffer to optimize script loading for Google Maps integration.
+	 *
+	 * This method adjusts the output buffer to include a preload tag for the script associated with
+	 * Google Maps if the script URL matches certain criteria. It also ensures the buffer is updated
+	 * for lazy loading and adds necessary filters.
+	 *
+	 * @param array $data An array containing the buffer and a modified flag.
+	 *                    - `buffer` (string): The HTML buffer to be modified.
+	 *                    - `modified` (bool): A flag indicating if the buffer was modified.
+	 *
+	 * @return array The modified `$data` array.
+	 */
 	public function w3tc_lazyload_mutator_before( $data ) {
 		$buffer = $data['buffer'];
 		$buffer = preg_replace_callback(
 			'~(<script\s[^>]+>)~i',
-			array( $this, 'tag_script' ), $buffer
+			array( $this, 'tag_script' ),
+			$buffer
 		);
 
-		if ( !empty( $this->preload_url ) ) {
+		if ( ! empty( $this->preload_url ) ) {
 			$preload_html = '<link rel="preload" href="' . esc_url( $this->preload_url ) . '" as="script">';
 
-			$buffer = preg_replace( '~<head(\s+[^>]*)*>~Ui',
-				'\\0' . $preload_html, $buffer, 1 );
+			$buffer = preg_replace(
+				'~<head(\s+[^>]*)*>~Ui',
+				'\\0' . $preload_html,
+				$buffer,
+				1
+			);
 
-			add_filter( 'w3tc_lazyload_on_initialized_javascript', array(
-				$this, 'w3tc_lazyload_on_initialized_javascript' ) );
+			add_filter( 'w3tc_lazyload_on_initialized_javascript', array( $this, 'w3tc_lazyload_on_initialized_javascript' ) );
 		}
 
-		$data['buffer'] = $buffer;
-		$data['modified'] |= !empty( $this->preload_url );
+		$data['buffer']    = $buffer;
+		$data['modified'] |= ! empty( $this->preload_url );
 
 		return $data;
 	}
 
-
-
+	/**
+	 * Processes a script tag to identify and handle specific Google Maps scripts.
+	 *
+	 * This method examines the script tag to check if it matches the Google Maps Easy plugin's
+	 * frontend script. If a match is found, it updates the `$preload_url` and removes the tag from the buffer.
+	 *
+	 * @param array $m Matches from the regular expression applied to the buffer.
+	 *                 - $m[0] (string): The full matched script tag.
```

Comment:
For "hash notation style", use `@type` instead of `@param`, otherwise, use `@param array $args {` to outline the array elements.  I like the latter.
https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#:~:text=1.1%20Parameters%20That%20Are%20Arrays

Reply:
Updated